### DISCUSSION TITLE
connection to external oracle base with user storage provider

### DISCUSSION THREAD

**Discussion by felipememphis** [keycloak/keycloak#15671] [Archive]
I'm using keycloak version 20.0.1 and I created a user storage provider that is connecting an external oracle base. I have the persistence.xml file and it connects to the Provicer class with session.getProvider(JpaConnectionProvider.class, "demo-jpa").getEntityManager(), only forced to use the persistence.xml to pass the hibernate.dialect only that we have several versions of oracle and I want to make it flexible. I couldn't find a way to dynamically pass or not use this persistence.xml file. Would there be any other way?

**Comment by dominiktopp** [comment]
There is a workaround (https://github.com/keycloak/keycloak/discussions/24170) but unfortunately no solution. Waiting for it, too.
